### ISSUE TITLE
Error when loading model with tf.keras.layers.Input() layer

### ISSUE THREAD

**Issue by andmax** [tensorflow/tensorflow#33357]
On TensorFlow 1.14 (OS Ubuntu 16.04), when I call tf.keras.models.load_model() to load an entire tf.Keras model (following https://www.tensorflow.org/tutorials/keras/save_and_load) if the model contains tf.keras.layers.Input(), the function fails with:

```
(...)
/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/saving/hdf5_format.py
in load_weights_from_hdf5_group(f, layers)
    735                      'containing ' + str(len(layer_names)) +
    736                      ' layers into a model with ' + str(len(filtered_layers)) +
--> 737                      ' layers.')
    738 
    739   # We batch weight value assignments in a single backend call

ValueError: You are trying to load a weight file containing 6 layers into a model with
0 layers.
```

The minimal code to reproduce is:

```
import tensorflow as tf
from tensorflow.keras.utils import to_categorical, HDF5Matrix

(Xtr, Ytr), (Xva, Yva) = tf.keras.datasets.cifar10.load_data()
Xtr, Ytr, Xva, Yva, nc = Xtr[:1000], Ytr[:1000], Xva[:100], Yva[:100], 10
Xtr, Xva = Xtr.astype('float32') / 255, Xva.astype('float32') / 255
Ytr, Yva, ins = to_categorical(Ytr, nc), to_categorical(Yva, nc), Xtr.shape[1:]

model = tf.keras.models.Sequential()
model.add(tf.keras.layers.Input(ins)) # (1)
model.add(tf.keras.layers.Conv2D(8, (3, 3))) # (2)
#model.add(tf.keras.layers.Conv2D(8, (3, 3), input_shape=ins)) # (3)
model.add(tf.keras.layers.Activation('relu'))
model.add(tf.keras.layers.Conv2D(8, (3, 3), activation='relu'))
model.add(tf.keras.layers.MaxPooling2D(pool_size=(2, 2)))
model.add(tf.keras.layers.Dropout(0.25))
model.add(tf.keras.layers.Conv2D(16, (3, 3), activation='relu'))
model.add(tf.keras.layers.Conv2D(16, (3, 3), activation='relu'))
model.add(tf.keras.layers.MaxPooling2D(pool_size=(2, 2)))
model.add(tf.keras.layers.Dropout(0.25))
model.add(tf.keras.layers.Flatten())
model.add(tf.keras.layers.Dense(32, activation='relu'))
model.add(tf.keras.layers.Dropout(0.5))
model.add(tf.keras.layers.Dense(nc, activation='softmax'))
opt = tf.keras.optimizers.RMSprop(lr=0.0001, decay=1e-6)
model.compile(loss='categorical_crossentropy', optimizer=opt, metrics=['acc'])

model.fit(x=Xtr, y=Ytr, batch_size=32, epochs=1, validation_data=(Xva, Yva))
model.save('model.hdf5')
del model
model = tf.keras.models.load_model('model.hdf5')
```

If I replace (1) and (2) by (3) the load_model() function works.

**Comment by oanush** [comment]
@andmax ,
Hi, can you please try using running the code in latest versions TF-[1.15rc3](https://colab.sandbox.google.com/drive/1B4O7P309eEi6GFTVfbeVThapeUyClEF0#scrollTo=ZgGEb__5B6Fz) and also [2.0](https://colab.sandbox.google.com/gist/oanush/be5e0dab0dc5c67e013cf51e881835a9/2-0.ipynb) i didn't face any issue. kindly find the gists for the same.Thanks!

**Comment by andmax** [comment]
Hi @oanush , I was able to run your gist 2.0 successfully but could not access the 1.15rc3.  But my question is: this bug is to be considered solved in TF 1.15 and 2.0?  So I should just upgrade from TF 1.14.

**Comment by oanush** [comment]
@andmax ,
Thank you checking on that, i have updated Tf-1.15rc3 gist, yes you can just upgrade to latest version 1.15rc3.

**Comment by andmax** [comment]
This issue has been fixed in TF 1.15 and 2.0.

**Comment by hardianlawi** [comment]
Hi, I encountered a similar issue when using `DenseFeatures` as my initial layer.
